### ISSUE TITLE
[no-extra-parens] false positive with type assertion

### ISSUE THREAD

**Issue by Dionysusnu** [typescript-eslint/typescript-eslint#1322]
**Repro**

rule not touched in .eslintrc rules property
```JSON
{
	"extends": [
		"eslint:recommended",
		"plugin:@typescript-eslint/eslint-recommended",
		"plugin:@typescript-eslint/recommended",
		"plugin:@typescript-eslint/recommended-requiring-type-checking"
	],
	"env": {
		"node": true,
		"es6": true
	},
	"parser": "@typescript-eslint/parser",
	"plugins": [
		"@typescript-eslint"
	],
}
```

this line, first parentheses is the error
`(msg as Message).edit(`Pong! ${(msg as Message).createdTimestamp - message.createdTimestamp}ms`);`
```TS
// relevant bit of the function
async execute(message: Message): Promise<void> {
		const msg = await message.channel.send('Pong!');
		(msg as Message).edit(`Pong! ${(msg as Message).createdTimestamp - message.createdTimestamp}ms`);
	},
```

**Expected Result**
No error, because the parentheses around `msg as Message` are necessary, otherwise the compiler would read `Message.edit()`, which throws the following compiler error
```
Cannot access 'Message.edit' because 'Message' is a type, but not a namespace. did you mean to retrieve the type of the property 'edit' in 'Message' with 'Message["edit"]'? 2713
```
**Actual Result**
ESlint error, saying "unnecessary" parentheses, while they aren't unnecessary because removing them causes a compiler error
```
9:3   error  Unnecessary parentheses around expression  no-extra-parens
```
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.11.0` |
| `@typescript-eslint/parser`        | `2.11.0` |
| `TypeScript`                       | `3.7.3` |
| `ESLint`                           | `6.7.2` |
| `node`                             | `10.16.0` |
| `pnpm`                              | `4.3.3` |

**Comment by bradzacher** [comment]
You will have to manually configure your lint config to use our version of the rule: `@typescript-eslint/no-extra-parens`.

I don't know what config has added the rule in, as `no-extra-parens` isn't recommended in any of the configs you have listed. You can debug via the `--print-config` argument if you like.
But it won't change the fact that the rule straight out of eslint base won't work ever!

**Comment by Dionysusnu** [comment]
Oops, I hadn't set it as a root config, and my normal javascript config was enabling the rule. Can't your recommended preset override the default setting and disable it, though?

**Comment by bradzacher** [comment]
> Can't your recommended preset override the default setting and disable it, though?

Sure, but that's only if we add the rule to our recommended set.  
However, it's a stylistic rule, so it is not recommended (we only recommend best practices, and there's too much heated discussion about stylistic preferences).

ESLint base does not have `no-extra-parens` in their recommended set either:
https://eslint.org/docs/rules/
![image](https://user-images.githubusercontent.com/7462525/70753598-09c54300-1d85-11ea-822f-178ea81e494d.png)

This is why we don't have an override.
